crates/typst-ide/src/complete.rs

```diff
@@ -1053,7 +1054,7 @@ impl<'a> CompletionContext<'a> {
 
     /// A small window of context before the cursor.
     fn before_window(&self, size: usize) -> &str {
-        &self.before[self.cursor.saturating_sub(size)..]
+        slice_at(self.before, self.cursor.saturating_sub(size)..self.before.len())
```

Comment:
`unscanny` already has the snapping functionality, so we could also do `Scanner::new(self.before).from(self.cursor.saturating_sub(size))`